src/Analytics/Adapter.php

```diff
@@ -25,6 +25,15 @@ abstract class Adapter
      */
     abstract public function getName(): string;
 
+    /**
+     * Adapter constructor.
+     * 
+     * @param string $configuration
+     * 
+     * @return Adapter
+     */
+    abstract public function __construct(string $configuration);
```

Comment:
As mentioned on the plausible PR. Let's avoid this "config" pattern.

Reply:
Agree. We did same in Utopia Logger and it only made it worse. Lets accept multiple variables with different data to prevent doing `explode(,)` later